tests/test_preprints.py

```diff
@@ -1806,6 +1814,41 @@ def test_highlighted_subjects(self):
         assert self.provider.has_highlighted_subjects is True
         assert set(self.provider.highlighted_subjects) == set([subj_aaa])
 
+    def test_change_preprint_provider_custom_taxonomies(self):
+        subject_two = SubjectFactory(provider=self.provider_two,
+            bepress_subject=self.subject_one.bepress_subject)
+        preprint = PreprintFactory(subjects=[[self.subject_one._id]], provider=self.provider_one, creator=self.user)
+        subject_problems = preprint.map_subjects_between_providers(self.provider_one, self.provider_two, self.auth)
+        preprint.refresh_from_db()
+        assert subject_problems == []
+        assert subject_two in preprint.subjects.all()
+
+    def test_change_preprint_provider_from_osf(self):
+        subject_two = SubjectFactory(provider=self.provider_one,
+            bepress_subject=self.subject_osf)
+        preprint = PreprintFactory(subjects=[[self.subject_osf._id]], provider=self.provider_osf, creator=self.user)
+        subject_problems = preprint.map_subjects_between_providers(self.provider_osf, self.provider_one, self.auth)
+        preprint.refresh_from_db()
+        assert subject_problems == []
+        assert subject_two in preprint.subjects.all()
+
+    def test_change_preprint_provider_to_osf(self):
+        subject_two = SubjectFactory(provider=self.provider_one,
+            bepress_subject=self.subject_osf)
+        preprint = PreprintFactory(subjects=[[subject_two._id]], provider=self.provider_one, creator=self.user)
+        subject_problems = preprint.map_subjects_between_providers(self.provider_one, self.provider_osf, self.auth)
+        preprint.refresh_from_db()
+        assert subject_problems == []
+        assert self.subject_osf in preprint.subjects.all()
+
+    def test_change_preprint_provider_problem_subject(self):
+        subject_two = SubjectFactory(provider=self.provider_one,
+            bepress_subject=self.subject_osf)
+        preprint = PreprintFactory(subjects=[[subject_two._id]], provider=self.provider_one, creator=self.user)
+        subject_problems = preprint.map_subjects_between_providers(self.provider_one, self.provider_two, self.auth)
+        preprint.refresh_from_db()
+        assert subject_problems == [subject_two.text]
+        assert subject_two in preprint.subjects.all()
```

Comment:
I'd add an admin app test for a user that is not a contributor on the preprint, but has `change_preprint` permissions .